src/ManagerZKit.ts

```diff
@@ -0,0 +1,127 @@
+import fs from "fs";
+import https from "https";
+import os from "os";
+import path from "path";
+import process from "process";
+import { v4 as uuid } from "uuid";
+
+import { ManagerZKitConfig, ManagerZKitPrivateConfig, PtauInfo, TemplateType } from "./types";
+
+export class ManagerZKit {
+  private _config: ManagerZKitPrivateConfig;
+
+  constructor(config: Partial<ManagerZKitConfig>) {
+    const projectRoot = process.cwd();
+
+    const ptau = config.ptau ? path.join(projectRoot, config.ptau) : path.join(os.tmpdir(), ".zkit/.ptau");
+
+    this._config = {
+      circuits: path.join(projectRoot, config.circuits ?? "circuits"),
+      artifacts: path.join(projectRoot, config.artifacts ?? "zkit-artifacts"),
+      verifiers: path.join(projectRoot, config.verifiers ?? "contracts/verifiers"),
+      ptau,
+      compiler: fs.readFileSync(require.resolve("@distributedlab/circom2/circom.wasm")),
+      templates: {
+        groth16: fs.readFileSync(path.join(__dirname, "templates", "verifier_groth16.sol.ejs"), "utf8"),
+      },
+    };
+  }
+
+  public async fetchPtauInfo(constraints: number): Promise<PtauInfo> {
+    const ptauId = Math.max(10, Math.ceil(Math.log2(constraints)));
+
+    if (ptauId > 20) {
```

Comment:
We should not restrict the user which schemes to compile, but prohibit the download of large files through our program